core/src/main/java/org/bitcoinj/evolution/AbstractQuorumState.java

```diff
@@ -201,7 +201,11 @@ public boolean reachedMaxFailedAttempts() {
     public abstract void requestUpdate(Peer peer, StoredBlock block);
 
     public void retryLastUpdate(Peer peer) {
-        peer.sendMessage(lastRequest.getRequestMessage());
+        if (peer != null) {
+            peer.sendMessage(lastRequest.getRequestMessage());
+        } else {
+            log.info("no peer supplied to retry the last update request: {}", lastRequest.request);
+        }
```

Comment:
We were getting an NPE if peer was null.  Not sure of the implications of doing this fix, but other changes should have a null peer if the DashJ has been completely disconnected from the network.  The sync process should resume when reconnecting.